src/common/services/session/session.service.js

```diff
@@ -46,95 +65,287 @@ const session = /* @ngInject */ function ($cookies, $rootScope, $http, $timeout,
+      $window.sessionStorage.removeItem(locationSearchOnLogin)
+      searchQueries.forEach((searchQuery) => {
+        const [search, value] = searchQuery.split('=')
+        if (search && value) {
+          $location.search(search, value)
+        }
+      })
+    }
+    removeOktaRedirectIndicator()
+    return $http({
+      method: 'POST',
+      url: oktaApiUrl('login'),
+      data: data,
+      withCredentials: true
+    })
+  }
+
+  async function createAccount (email, firstName, lastName, isTest = false) {
+    const isAuthenticated = await authClient.isAuthenticated()
+    if (currentRole() !== Roles.public || isAuthenticated) {
+      const email = isAuthenticated && (await authClient.getUser()).email
+      return {
+        status: 'error',
+        data: [`Already logged in to Okta${email ? ` with email: ${email}` : ''}. You will be redirected to the Sign In page in a few seconds.`, 'Another Error'],
+        redirectToSignIn: true
+      }
+    }
+
+    const data = { }
+
+    if (angular.isDefined(email)) data.email = email
+    if (angular.isDefined(firstName)) data.first_name = firstName
+    if (angular.isDefined(lastName)) data.last_name = lastName
+    const dataAsString = JSON.stringify(data)
+    try {
+      const createAccount = await $http({
+        method: 'POST',
+        url: oktaApiUrl('create'),
+        data: data,
         withCredentials: true
       })
+
+      $cookies.put(
+        createAccountDataCookieName,
+        dataAsString,
+        {
+          path: '/',
+          domain: isTest ? '' : cookieDomain,
+          expires: moment().add(2, 'hours').toISOString()
+        }
       )
+      return {
+        status: 'success',
+        data: createAccount
+      }
+    } catch (error) {
+      return await catchCreateAccountErrors(error, dataAsString, email, isTest)
+    }
   }
 
-  function signUp (email, password, first_name, last_name) {
-    // https://github.com/CruGlobal/cortex_gateway/wiki/Create-User
-    return Observable
-      .from($http({
-        method: 'POST',
-        url: casApiUrl('/register'),
-        withCredentials: true,
-        data: {
-          email: email,
-          password: password,
-          firstName: first_name,
-          lastName: last_name
+  async function catchCreateAccountErrors (errorObject, dataAsString, email, isTest) {
+    try {
+      if (errorObject.status === 401) {
+        throw new Error(errorObject.message)
+      }
+      const errors = errorObject?.data?.error
+        ? errorObject.data.error.split(',')
+            .filter((str) => str.includes(':errorSummary=>') && !str.includes('Api validation failed: login'))
+            .map((str) => str.match(/"([^"]+)"/)[1].replace(/["]/g, ''))
+        : errorObject
+
+      let checkIfAccountIsPending = false
+
+      const formattedErrors = errors.map((error) => {
+        switch (error) {
+          case 'login: An object with this field already exists in the current organization':
```

Comment:
Could we do a quick test with a different browser language setting? Wondering if there is any chance that Okta will translate these error messages in that case and bypass our nice error handling.

Reply:
The errors will always be in English as the service account that created the API token has the display language setting set to English.

https://support.okta.com/help/s/article/How-to-change-the-display-language-of-the-error-that-you-receive-trough-API-calls?language=en_US